consensus/src/quorum/verifiers.rs

```diff
@@ -25,127 +25,79 @@ use crate::user::committee::{Committee, CommitteeSet};
+    ch: &ConsensusHeader,
     vote: &Vote,
     sv: &StepVotes,
     committees_set: &RwLock<CommitteeSet<'_>>,
     seed: Seed,
     step: StepName,
-) -> Result<(QuorumResult, Vec<Voter>), StepSigError> {
-    let round = header.round;
-    let iteration = header.iteration;
-
-    let mut exclusion_list = vec![];
-    let generator = committees_set
-        .read()
-        .await
-        .provisioners()
-        .get_generator(iteration, seed, round);
-
-    exclusion_list.push(generator);
-
-    if exclude_next_generator(iteration) {
-        let next_generator = committees_set
-            .read()
-            .await
-            .provisioners()
-            .get_generator(iteration + 1, seed, round);
-
-        exclusion_list.push(next_generator);
+) -> Result<Vec<Voter>, StepSigError> {
+    // When verifying a NoQuorum Attestation, the Validation StepVotes should be
+    // empty. To be on the safe side, we simply skip verification, instead of
+    // failing verification
+    if step == StepName::Validation && *vote == Vote::NoQuorum {
+        return Ok(vec![]);
     }
 
-    let cfg =
-        sortition::Config::new(seed, round, iteration, step, exclusion_list);
-
-    if committees_set.read().await.get(&cfg).is_none() {
-        let _ = committees_set.write().await.get_or_create(&cfg);
-    }
-
-    let set = committees_set.read().await;
-    let committee = set.get(&cfg).expect("committee to be created");
-
-    let (quorum_result, voters) =
-        verify_votes(header, step, vote, sv, committee)
-        .map_err(|e|
-            {
-                error!( "invalid {:?}, vote = {:?}, round = {}, iter = {}, seed = {}, sv = {:?}, err = {}",
-                    step,
-                    vote,
-                    header.round,
-                    header.iteration,
-                    to_str(seed.inner()),
-                    sv,
-                    e
-                );
-                e
-            }
-        )?;
-
-    Ok((quorum_result, voters))
-}
-
-pub struct QuorumResult {
-    pub total: usize,
-    pub target_quorum: usize,
+    let committee = get_step_committee(ch, committees_set, seed, step).await;
+
+    // Verify the aggregated signature is valid and reach the quorum threshold
+    let voters =
+        verify_quorum_votes(ch, step, vote, sv, &committee).map_err(|e| {
+            error!(
+                event = "Invalid StepVotes",
+                reason = %e,
+                ?vote,
+                round = ch.round,
+                iter = ch.iteration,
+                ?step,
+                seed = to_str(seed.inner()),
+                ?sv
+            );
+
+            e
+        })?;
+
+    Ok(voters)
 }
 
-impl QuorumResult {
-    pub fn quorum_reached(&self) -> bool {
-        self.total >= self.target_quorum
-    }
-}
-
-pub fn verify_votes(
+pub fn verify_quorum_votes(
     header: &ConsensusHeader,
     step: StepName,
     vote: &Vote,
-    step_votes: &StepVotes,
+    sv: &StepVotes,
```

Comment:
nit again: I'd leave `step_votes` rather than `sv` here.

Reply:
I changed it bc the StepVotes argument is called `sv` in other functions. But I can also change all instances to be `step_votes`

Reply:
I eliminated the use of `sv` in the code in favor of `step_votes`